packages/lib/src/data-grid/DataGrid.tsx

```diff
@@ -0,0 +1,329 @@
+            return (
+              <HierarchyContainer level={row.rowLevel || 0}>
+                {renderHierarchyTrigger(rowsToRender, row, uniqueRowId, firstColumnKey, (rows) =>
+                  setRowsToRender(rows)
+                )}
+              </HierarchyContainer>
+            );
+          }
+          return (
+            <HierarchyContainer level={row.rowLevel || 0} className="ellipsis-cell">
+              {row[firstColumnKey]}
+            </HierarchyContainer>
+          );
+        },
+      };
+    }
+    if (selectable) {
+      expectedColumns = [
+        {
+          key: "selected",
+          name: "",
+          maxWidth: 50,
+          width: 50,
+          renderCell({ row }) {
+            if (!row.isExpandedChildContent) {
+              return (
+                <ActionContainer>{renderCheckbox(rows, row, uniqueRowId, selectedRows, onSelectRows)}</ActionContainer>
+              );
+            }
+          },
+          renderHeaderCell: () => {
+            return (
+              <ActionContainer>{renderHeaderCheckbox(rows, uniqueRowId, selectedRows, onSelectRows)}</ActionContainer>
+            );
+          },
+        },
+        ...expectedColumns,
+      ];
+    }
+    return expectedColumns;
+  }, [selectable, expandable, columns, rowsToRender, onSelectRows, rows, summaryRow, uniqueRowId, selectedRows]);
+  // array with the order of the columns
+  const [columnsOrder, setColumnsOrder] = useState((): number[] => columnsToRender.map((_, index) => index));
+  const [sortColumns, setSortColumns] = useState<readonly SortColumn[]>([]);
+  // const [visibleColumns, setVisibleColumns] = useState(
+  //   columnsNamesIntoOptions(columns).map((visibleColumn) => {
+  //     return visibleColumn.value;
+  //   })
+  // );
+
+  useEffect(() => {
+    setColumnsOrder(columnsToRender.map((_, index) => index));
+  }, [columnsToRender]);
+
+  const reorderedColumns = useMemo(() => {
+    // Array ordered by columnsOrder
+    return columnsOrder.map((index) => columnsToRender[index]);
+  }, [columnsOrder, columnsToRender]);
+
+  const onColumnsReorder = (sourceKey: string, targetKey: string) => {
+    setColumnsOrder((columnsOrder) => {
+      const sourceColumnOrderIndex = columnsOrder.findIndex((index) => columnsToRender[index].key === sourceKey);
+      const targetColumnOrderIndex = columnsOrder.findIndex((index) => columnsToRender[index].key === targetKey);
+      const newColumnsOrder = columnsOrder.slice();
+      newColumnsOrder.splice(sourceColumnOrderIndex, 1);
+      newColumnsOrder.splice(targetColumnOrderIndex, 0, columnsOrder[sourceColumnOrderIndex]);
+      return newColumnsOrder;
+    });
+  };
+
+  const onRowsChange = (newRows: { [key: string]: string | number | React.ReactNode }[]) => {
+    // call function to change rows, like when they have been edited
+    console.log("new rows: ");
+    console.log(newRows);
+  };
+
+  const sortedRows = useMemo((): readonly GridRow[] => {
```

Comment:
Is it really needed to have a separate object for `sortedRows` instead of applying a sort to the `rowsToRender` themselves?

Reply:
I find it easier to understand this way, by using a useMemo we have it isolated to the rest of the functionality and it is also recalculated if the sorting changes.